pkg/deviceplugin/api.go

```diff
@@ -58,13 +58,20 @@ func NewDeviceInfo(state string, nodes []pluginapi.DeviceSpec, mounts []pluginap
 		cdiSpec:     cdiSpec,
 	}
 
-	devPaths := []string{}
+	var devPathsComputed []string
 
-	for _, node := range nodes {
-		devPaths = append(devPaths, node.HostPath)
+	// If devPaths are provided, use them; otherwise, generate from nodes
+	if len(devPaths) > 0 && devPaths[0] != "" {
+		devPathsComputed = devPaths
```

Comment:
Can you explain why this is needed in general?

Reply:
This change suppresses error messages that occur when the code references the real /dev/char directory, which lacks the generated devices major:minor info. If devPath is not provided, the function behaves as before. In essence, the error is not critical because fakedri pods are not expected to access /dev/char.

Reply:
Previously fakedev support used different sysfs and devfs directories. Can those be used to again to expose the fake dev-devices?

In general, I'm against this particular change as it changes the internal API with no real added functionality.

Reply:
"Previously fakedev support used different sysfs and devfs directories" These are still used. I was actually looking also option that fake devs would have been mounted to /dev /sys but turn out to be difficult and required init-container and never got that even working that way. 

Changes the internal API is not needed. I haven't got that error anymore, I have tested fakedri now in big cluster 250 nodes with nfd overlays and not got this error anymore in any condition I guess that was false diagnosis before.
api.go is now unchanged.